operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,1082 @@
+                    .expect("Data vector length should match the number of pixels in the tile");
+
+                Ok(RasterTile2D::new_with_tile_info(
+                    query.time_interval,
+                    tile_info,
+                    GridOrEmpty::Grid(tile_grid.into()),
+                ))
+            });
+            Ok(tiles.boxed())
+        } else {
+            Ok(generate_zeroed_tiles(self.tiling_specification, query))
+        }
+    }
+}
+
+pub struct DensityRasterizationQueryProcessor {
+    input: TypedVectorQueryProcessor,
+    tiling_specification: TilingSpecification,
+    radius: f64,
+    stddev: f64,
+}
+
+#[async_trait]
+impl RasterQueryProcessor for DensityRasterizationQueryProcessor {
+    type RasterType = f64;
+
+    /// Performs a gaussian density rasterization.
+    /// For each tile, the spatial bounds are extended by `radius` in x and y direction.
+    /// All points within these extended bounds are then queried. For each point, the distance to
+    /// its surrounding tile pixels (up to `radius` distance) is measured and input into the
+    /// gaussian density function with the configured standard deviation. The density values
+    /// for each pixel are then summed to result in the tile pixel grid.
+    async fn raster_query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> util::Result<BoxStream<'a, util::Result<RasterTile2D<Self::RasterType>>>> {
+        if let MultiPoint(points_processor) = &self.input {
+            let tiling_strategy = self
+                .tiling_specification
+                .strategy(query.spatial_resolution.x, -query.spatial_resolution.y);
+
+            let tile_size_x = tiling_strategy.tile_size_in_pixels.axis_size_x();
+            let tile_size_y = tiling_strategy.tile_size_in_pixels.axis_size_y();
+
+            let tiles = stream::iter(
+                tiling_strategy.tile_information_iterator(query.spatial_bounds),
+            )
+            .then(move |tile_info| async move {
+                let tile_bounds = tile_info.spatial_partition();
+
+                let vector_query = VectorQueryRectangle {
+                    spatial_bounds: extended_bounding_box_from_spatial_partition(
+                        tile_bounds,
+                        self.radius,
+                    ),
+                    time_interval: query.time_interval,
+                    spatial_resolution: query.spatial_resolution,
+                };
+
+                let tile_geo_transform = tile_info.tile_geo_transform();
+
+                let mut chunks = points_processor.query(vector_query, ctx).await?;
+
+                let mut tile_data = vec![0.; tile_size_x * tile_size_y];
+
+                while let Some(chunk) = chunks.next().await {
+                    let chunk = chunk?;
+                    let (send, mut receive) = tokio::sync::mpsc::unbounded_channel();
+                    ctx.thread_pool().install(|| {
```

Comment:
Please use `spawn_blocking_with_thread_pool` here since this is a blocking computation.

Reply:
https://github.com/geo-engine/geoengine/pull/683/commits/1bb2b3ec81799658c4f72d696aec5079bf1ac9e9